crates/api/src/data_structures.rs

```diff
@@ -0,0 +1,26 @@
+pub(crate) mod ir {
+    pub(crate) use cranelift_codegen::ir::{types, AbiParam, ArgumentPurpose, Signature, Type};
+}
+
+pub(crate) mod settings {
+    pub(crate) use cranelift_codegen::settings::{builder, Flags};
+}
```

Comment:
Can `settings` be moved into the `trampoline` mod?

Reply:
Yes, the `settings` and `Flags` are required during api::Config construction.